Makefile

```diff
@@ -0,0 +1,52 @@
+#
+# Makefile for mbed Client C++ Library
+#
+# List of subdirectories to build
+TEST_FOLDER := ./test/
+
+# Define compiler toolchain with CC or PLATFORM variables
+# Example (GCC toolchains, default $CC and $AR are used)
+# make
+#
+# OR (Cross-compile GCC toolchain)
+# make PLATFORM=arm-linux-gnueabi-
+#
+# OR (ArmCC/Keil)
+# make CC=ArmCC AR=ArmAR
+#
+# OR (IAR-ARM)
+# make CC=iccarm
+
+LIB = libmbed-trace.a
+
+# List of unit test directories for libraries
+UNITTESTS := $(sort $(dir $(wildcard $(TEST_FOLDER)*/utest/*)))
+
+
+include sources.mk
+include include_dirs.mk
+
+SERVLIB_DIR := ../libService
```

Comment:
mbed-trace doesnt have libservice dependecy as long as it use internal ns-list. I think we should avoid libservice dependency here.